### ISSUE TITLE
Runtime Error: undefined symbol: _ZdaPvm, version Qt_5

### ISSUE THREAD

**Issue by liushuya7** [jhu-saw/sawNDITrackerROS2#1]
Hi @adeguet1,

I encountered an issue:
`colcon build` is successful. However, when I do `ros2 run ndi_tracker ndi_tracker`, it gives this error:
```bash
symbol lookup error: $COLCON_WORKSPACE/install/ndi_tracker/lib/ndi_tracker/ndi_tracker: undefined symbol: _ZdaPvm, version Qt_5
```

My Enviroment:
- Kubuntu 20.04
- Qt 5.12.12
- with **cisst** and **sawNDItracker** pre-built

Is this because the Qt version?

**Comment by adeguet1** [comment]
Can you locate the executable `ndi_tracker` (based on your ticket, it's under your workspace and then `install/ndi_tracker/lib/ndi_tracker`) and do `ldd ndi_tracker`?   This will show which libraries are being used.  I suspect the Qt libraries loaded are not the system libraries but maybe some libraries installed for another program (maybe Matlab or virtualenv or pip).  If this is the case, you will have to make sure that ROS node is started in an environment that doesn't redefine the PATH or LD_LIBRARY_PATH.

To note, for the dVRK packages, I've figured out how to get all of cisst and SAW compiled in the colcon workspace instead of building and installing `.deb` packages.  I think this is a much better solution since it prevents installing libraries system wide.  I will patch the sawNDITracker packages to allow this but I don't think this would solve your issue.

**Comment by liushuya7** [comment]
Hi @adeguet1,

I think you are right. Because I am not using the system Qt libraries. I downloaded and installed Qt, and I have added it to my environment path `export PATH=/home/josh/Qt/5.12.12/gcc_64/bin:$PATH`.
When do I `ldd ndi_tracker,` I realize `ndi_tracker` was built still using the system Qt libraries.
```bash
	linux-vdso.so.1 (0x00007ffe99a62000)
	libjsoncpp.so.1 => /lib/x86_64-linux-gnu/libjsoncpp.so.1 (0x00007f897469c000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f8974679000)
	libGL.so.1 => /lib/x86_64-linux-gnu/libGL.so.1 (0x00007f89745f1000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f89745eb000)
	libQt5Widgets.so.5 => /lib/x86_64-linux-gnu/libQt5Widgets.so.5 (0x00007f8973f50000) <<<---
	libQt5Gui.so.5 => /lib/x86_64-linux-gnu/libQt5Gui.so.5 (0x00007f8973963000) <<<---
	libQt5Core.so.5 => /lib/x86_64-linux-gnu/libQt5Core.so.5 (0x00007f897341a000) <<<---
	libcrtk_msgs__rosidl_typesupport_cpp.so => not found
	libcisst_msgs__rosidl_typesupport_cpp.so => not found
	... ...
```

**Comment by liushuya7** [comment]
Thanks! This is solved by adding `/home/josh/Qt/5.12.12/gcc_64/lib/` to the `LD_LIBRARY_PATH`.
